pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -631,3 +615,96 @@ func TestGetPodCIDRsOnNode(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoveStaleGatewayRoutes(t *testing.T) {
+	fakeClientSet := fake.NewSimpleClientset(node1)
+	c := newController(t, &config.NetworkConfig{}, fakeClientSet)
+	defer c.queue.ShutDown()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+
+	c.routeClient.EXPECT().Reconcile([]string{podCIDR.String()})
+	err := c.removeStaleGatewayRoutes()
+	assert.NoError(t, err)
+}
+
+func TestRemoveStaleWireGuardPeers(t *testing.T) {
+	nodeX := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "nodeX",
+			Annotations: map[string]string{types.NodeWireGuardPublicAnnotationKey: "fakekey"},
+		},
+	}
+	fakeClientSet := fake.NewSimpleClientset(nodeX)
+	c := newController(t, &config.NetworkConfig{
+		TrafficEncryptionMode: config.TrafficEncryptionModeWireGuard,
+	}, fakeClientSet)
+	defer c.queue.ShutDown()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+
+	err := c.removeStaleWireGuardPeers()
+	assert.NoError(t, err)
+}
+
+func TestDeleteNodeRoute(t *testing.T) {
+	fakeClientSet := fake.NewSimpleClientset(node1)
+	tests := []struct {
+		name string
+		mode config.TrafficEncryptionModeType
+	}{
+		{
+			name: "delete a Node with IPSec mode",
+			mode: config.TrafficEncryptionModeIPSec,
+		},
+		{
+			name: "delete a Node with WireGuard mode",
+			mode: config.TrafficEncryptionModeWireGuard,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := newController(t, &config.NetworkConfig{
+				TrafficEncryptionMode: tt.mode,
+			}, fakeClientSet)
+			c.installedNodes.Add(&nodeRouteInfo{
+				nodeName: node1.Name,
+				podCIDRs: []*net.IPNet{podCIDR},
+			})
+			defer c.queue.ShutDown()
+
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			c.informerFactory.Start(stopCh)
+			c.informerFactory.WaitForCacheSync(stopCh)
+
+			if tt.mode == config.TrafficEncryptionModeIPSec {
+				c.interfaceStore.AddInterface(&interfacestore.InterfaceConfig{
```

Comment:
Case specific preparation and expectation should be set in the case table, instead of the case general code.
```
tests := []struct {
	name string
	mode config.TrafficEncryptionModeType
	interfaces []interfacestore.InterfaceConfig
	expectedCalls func(mockOFClient *openflowtest.MockClient, mockRouteClient *routetest.MockInterface, ...)
}{
	{
		name: "delete a Node with IPSec mode",
		mode: config.TrafficEncryptionModeIPSec,
		interfaces: ...
		expectedCalls: 
	},
	{
		name: "delete a Node with WireGuard mode",
		mode: config.TrafficEncryptionModeWireGuard,
		expectedCalls: 
	},
}
```

Reply:
Done